File: Implementation-Algo-books/map_direction.cpp
#include<iostream>
#include<algorithm>
#include<vector>
#include<string>
#include<sstream>
#include<ctime>
using namespace std;

int main()
{
	//time_t start_time = time(0);

	int size = 0;
	string input;
	string token;
	vector<string> movement;
	const char delimiter = ' ';

	cin >> size;
	cin.ignore();
	getline(cin, input);
	stringstream ss(input);

	while (getline(ss, token, delimiter))
	{
		movement.push_back(token);
	}

	//map 과 관련되어서 변수 선언
	int x = 1, y = 1;
	for (int i = 0; i < movement.size(); i++)
	{
		if (movement[i].compare("L") == 0)
		{
			if (x > 1) x -= 1;
		}
		else if (movement[i].compare("R") == 0)
		{
			if (x < size) x += 1;
		}
		else if (movement[i].compare("U") == 0)
		{
			if (y > 1) y -= 1;
		}
		else
		{
			if (y < size) y += 1;
		}
	}

	cout << y << ' ' << x << endl;
	//time_t end_time = time(0);

	//cout << "time : " << end_time - start_time;

	return 0;


}

File: python/implementation_algorithms/map_direction.py
# 시간 체크
import time
start_time = time.time()
# 입력 받기
n = int(input())
plans = input().split()
x, y, nx, ny = 1, 1, 1, 1

# 이동 경로 선언
dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
move_types = ['L', 'R', 'U', 'D']

for plan in plans:

    for i in range(len(move_types)):
        if plan == move_types[i]:
            nx = x + dx[i]
            ny = y + dy[i]

    if nx < 1 or nx > n or ny < 1 or ny > n:
        continue

    x, y = nx, ny

print(x, y)
end_time = time.time()
print("time: ", end_time-start_time)
